js/client/modules/@arangodb/test-helper.js

```diff
@@ -0,0 +1,169 @@
+/*jshint strict: false */
+/*global arango, db */
+
+// //////////////////////////////////////////////////////////////////////////////
+// / @brief Helper for JavaScript Tests
+// /
+// / @file
+// /
+// / DISCLAIMER
+// /
+// / Copyright 2010-2012 triagens GmbH, Cologne, Germany
+// /
+// / Licensed under the Apache License, Version 2.0 (the "License")
+// / you may not use this file except in compliance with the License.
+// / You may obtain a copy of the License at
+// /
+// /     http://www.apache.org/licenses/LICENSE-2.0
+// /
+// / Unless required by applicable law or agreed to in writing, software
+// / distributed under the License is distributed on an "AS IS" BASIS,
+// / WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// / See the License for the specific language governing permissions and
+// / limitations under the License.
+// /
+// / Copyright holder is triAGENS GmbH, Cologne, Germany
+// /
+// / @author Wilfried Goesgens
+// / @author Copyright 2011-2012, triAGENS GmbH, Cologne, Germany
+// //////////////////////////////////////////////////////////////////////////////
+
+let internal = require('internal'); // OK: processCsvFile
+let { 
+  getServerById,
+  getServersByType,
+  getEndpointById,
+  getEndpointsByType,
+  Helper,
+  deriveTestSuite,
+  deriveTestSuiteWithnamespace,
+  typeName,
+  isEqual,
+  compareStringIds,
+    } = require('@arangodb/test-helper-common');
+
+exports.getServerById = getServerById;
+exports.getServersByType = getServersByType;
+exports.getEndpointById = getEndpointById;
+exports.getEndpointsByType = getEndpointsByType;
+exports.Helper = Helper;
+exports.deriveTestSuite = deriveTestSuite;
+exports.deriveTestSuiteWithnamespace = deriveTestSuiteWithnamespace;
+exports.typeName = typeName;
+exports.isEqual = isEqual;
+exports.compareStringIds = compareStringIds;
+
+const primaryEndpoint = arango.getEndpoint();
```

Comment:
I don't think the endpoint should be retrieved once per script, but at the beginning of each function that temporarily modifies it.
Otherwise all functions may change back to an already outdated endpoint, e.g. if more than one server is contacted.